### PR TITLE
TriBITS snapshot on 6/7/2019 to bring in Trilinos_ENABLE_INSTALLATION_TESTING documentation (#5097)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
Brings in update of TriBITS that includes documentation for `Trilinos_ENABLE_INSTALLATION_TESTING` to help support setting up Trilinos installation testing (#5027) another with other stuff pushed to TriBITS (see [here](http://trac.trilinos.org/wiki/TriBITSTrilinosDev#tracing_snapshot_commits) for how to see what changed).

This should allow the @trilinos/framework team to set up installation testing for Trilinos.  With the TribitsCTestDriver.cmake code, one could do this pretty quickly.

NOTE: This update does not touch tribits/core/ or even tribits/ctest_driver/.  It only touches tribits/ci_support/, tribits/doc/, tribits/examples/ and tribits/python_utils/.  And even the change to tribits/ci_support/ is only changing the CDashQueryAnalyzeReport.py module.

```
Origin repo remote tracking branch: 'github/master'
Origin repo remote repo URL: 'github = git@github.com:TriBITSPub/TriBITS.git'

At commit:

commit c8b8ef0483f588372c380ee18fcb153e27ddb7d9
Author:  Roscoe A. Bartlett <rabartl@sandia.gov>
Date:    Fri Jun 7 17:57:40 2019 -0600
Summary: Merge branch '5027-enable-installation-testing' (trilinos/Trilinos#5027)
```

### CODE REVIEW

File: cmake/tribits/ci_support/CDashQueryAnalyzeReport.py [hunk 1]
```diff
@@ -1055,8 +1055,8 @@ def dateFromBuildStartTime(buildStartTime):
 #
 # Inputs:
 #
-#   testHistoryLOD [in]: List of test dicts for the same test.  This input
-#   list nore its elements are modified in this call.  The base list object is
+#   testHistoryLOD [in]: List of test dicts for the same test.  Neigher this
```
Comment (mhoemmen) [inline cmake/tribits/ci_support/CDashQueryAnalyzeReport.py]: "Neither"

File: cmake/tribits/doc/build_ref/TribitsBuildReferenceBody.rst [hunk 2]
```diff
@@ -3347,6 +3347,58 @@ This will ensure that every package that builds correctly will get installed.
 (The default 'install' target aborts on the first file install failure.)
 
 
+Installation Testing
+====================
+
+The CMake project <Project> has built-in support for testing an installation
+of itself using its own tests and examples.  The way it works is to configure,
+build, and install just the libraries and header files using::
```
Comment (mhoemmen) [inline cmake/tribits/doc/build_ref/TribitsBuildReferenceBody.rst]: Does double colon at the end of a line mean something in Markdown?  I see it a couple times in this PR.

File: cmake/tribits/doc/build_ref/TribitsBuildReferenceBody.rst [hunk 3]
```diff
@@ -3347,6 +3347,58 @@ This will ensure that every package that builds correctly will get installed.
...
+    -D<Project>_ENABLE_ALL_PACKAGES=ON \
+    -D<Project>_ENABLE_TESTS=OFF \
+    [other options] \
+    <projectDir>
+
+  $ make -j16 install   # or ninja -j16
+
+and then create a different build directory to configure and build just the
+tests and examples (not the libraries) against the pre-installed libraries and
+header files using::
```
Comment (mhoemmen) [inline cmake/tribits/doc/build_ref/TribitsBuildReferenceBody.rst]: another double colon (see above)

File: cmake/tribits/python_utils/gitdist.py [hunk 4]
```diff
@@ -438,14 +441,18 @@ def getHelpTopicsStr():
...
 
   $ alias gitdist-status="gitdist dist-repo-status"
   $ alias gitdist-mod="gitdist --dist-mod-only"
   $ alias gitdist-mod-status="gitdist dist-repo-status --dist-mod-only"
+  $ alias gitdist-repo-versions="gitdist --dist-no-color log -1 \
+    --pretty=format:\"%h [%ad] <%ae>%n%s\" | grep -v \"^$\""
 
-(i.e. add these to your ~/.bash_profile file.)
+These are added by sourcing the provided file 'gitdist-setup.sh' as well as
+some useful commandline tab completion (which be sourced in you
```
Comment (mhoemmen) [inline cmake/tribits/python_utils/gitdist.py]: "which should be sourced in your"?

File: cmake/tribits/doc/build_ref/TribitsBuildReferenceBody.rst (hunk 2 above)
Comment (bartlettroscoe) [reply cmake/tribits/doc/build_ref/TribitsBuildReferenceBody.rst]: @mhoemmen asked:

> Does double colon at the end of a line mean something in Markdown? I see it a couple times in this PR.

This is reStructuredText (RST), not MarkDown.  In RST, the double '::' means to show in as raw text.  This is equivalent to using:

```
  raw text
```

in Markdown.

File: cmake/tribits/ci_support/CDashQueryAnalyzeReport.py (hunk 1 above)
Comment (bartlettroscoe) [reply cmake/tribits/ci_support/CDashQueryAnalyzeReport.py]: Fixed in TriBITSPub/TriBITS@d96f8351703a595524e5ebd76c14e5e6ba49754a.  Thanks!

File: cmake/tribits/python_utils/gitdist.py (hunk 4 above)
Comment (bartlettroscoe) [reply cmake/tribits/python_utils/gitdist.py]: Fixed in TriBITSPub/TriBITS@d96f8351703a595524e5ebd76c14e5e6ba49754a.  Thanks!

File: cmake/tribits/doc/build_ref/TribitsBuildReferenceBody.rst (hunk 3 above)
Comment (bartlettroscoe) [reply cmake/tribits/doc/build_ref/TribitsBuildReferenceBody.rst]: See above

File: cmake/tribits/doc/build_ref/TribitsBuildReferenceBody.rst (hunk 2 above)
Comment (mhoemmen) [reply cmake/tribits/doc/build_ref/TribitsBuildReferenceBody.rst]: Thanks Ross!

**Pull request merged**